sycl/source/detail/ur.cpp

```diff
@@ -113,8 +113,10 @@ static void initializePlugins(std::vector<PluginPtr> &Plugins,
   // enable full validation by default.
   if(!LoaderConfig) {
     CHECK_UR_SUCCESS(urLoaderConfigCreate(&LoaderConfig))
-    CHECK_UR_SUCCESS(urLoaderConfigEnableLayer(LoaderConfig,
-                                               "UR_LAYER_PARAMETER_VALIDATION"))
+    if (std::getenv("UR_LOG_VALIDATION")) {
```

Comment:
This is certainly a way of doing this. Ideally validation would be enabled in CI, though, to stop invalid API usage issues slipping through.

Reply:
Do you have a different method to enable this in CI but not in normal runtime?

Reply:
Path of least resistance, based on this approach, would be to update the GitHub Actions workflows to set `UR_LOG_VALIDATION` during testing.

My personal preference would be to have this enabled by default and disabled by a CMake flag but that has it's own drawbacks in that release process would need to be updated to disabled validation by default.

Reply:
The validation layer can also be enabled via `UR_ENABLE_LAYERS=UR_LAYER_PARAMETER_VALIDATION` which is separate from `UR_LOG_VALIDATION` which will output additional info.